sizedwaitgroup.go

```diff
@@ -82,3 +82,20 @@ func (s *SizedWaitGroup) Done() {
 func (s *SizedWaitGroup) Wait() {
 	s.wg.Wait()
 }
+
+// Wait blocks until the SizedWaitGroup counter is zero or the context is Done.
+// See sync.WaitGroup documentation for more information.
+func (s *SizedWaitGroup) WaitWithContext(ctx context.Context) error {
+	done := make(chan struct{})
+	go func() {
+		defer close(done)
+		s.Wait()
+		done <- struct{}{}
+	}()
```

Comment:
this will leak goroutine each time `WaitWithContext` is called w/ an expired context until wg finishes.

Reply:
Could document it as such?  Given a context that has been cancelled, via os.Interrupt for example, I thought it better to orphan the goroutine and have the application exit when graceful immediate shutdown of all the goroutines in the sizedWaitGroup is not possible.